src/mpc/scenario.jl

```diff
@@ -76,32 +106,117 @@ function MPCScenario(d::Dict)
     if haskey(d, "ElectricStorage")
-        # only modeling electrochemical storage so far
         storage_dict = Dict(dictkeys_tosymbols(d["ElectricStorage"]))
     else
         storage_dict = Dict(:size_kw => 0.0, :size_kwh => 0.0)
     end
-    storage = Storage(Dict{String, AbstractStorage}("ElectricStorage" => MPCElectricStorage(; storage_dict...)))
+    storage_structs["ElectricStorage"] = MPCElectricStorage(; storage_dict...)
 
-    electric_load = MPCElectricLoad(; dictkeys_tosymbols(d["ElectricLoad"])...)
+    if haskey(d, "HydrogenStorageLP")
+        storage_structs["HydrogenStorageLP"] = MPCHydrogenStorageLP(; dictkeys_tosymbols(d["HydrogenStorageLP"])...)
+    end
+    if haskey(d, "HydrogenStorageHP")
+        storage_structs["HydrogenStorageHP"] = MPCHydrogenStorageHP(; dictkeys_tosymbols(d["HydrogenStorageHP"])...)
+    end
+
+    if haskey(d, "HighTempThermalStorage")
+       storage_structs["HighTempThermalStorage"] = MPCHighTempThermalStorage(; dictkeys_tosymbols(d["HighTempThermalStorage"])...)
+    end
 
-    electric_tariff = MPCElectricTariff(d["ElectricTariff"])
+    storage = Storage(storage_structs)
+   
+    if !(settings.off_grid_flag)
+        electric_tariff = MPCElectricTariff(d["ElectricTariff"])
+    else
+        tariff_dict = Dict([("energy_rates", zeros(length(electric_load.loads_kw)))])
+        electric_tariff = MPCElectricTariff(tariff_dict)
+    end
+
+    if haskey(d, "Wind")
+        wind = MPCWind(; dictkeys_tosymbols(d["Wind"])...)
+    else
+        wind = MPCWind(; size_kw=0)
+    end
 
     if haskey(d, "Generator")
         generator = MPCGenerator(; dictkeys_tosymbols(d["Generator"])...)
     else
         generator = MPCGenerator(; size_kw=0)
     end
 
+    if haskey(d, "ProcessHeatLoad")
+        process_heat_load = MPCProcessHeatLoad(; dictkeys_tosymbols(d["ProcessHeatLoad"])...)
+    else
+        process_heat_load = MPCProcessHeatLoad()
+    end
+
+    if haskey(d, "ElectricHeater")
+        electric_heater = MPCElectricHeater(; dictkeys_tosymbols(d["ElectricHeater"])...)
+    else
+        electric_heater = MPCElectricHeater(; size_mmbtu_per_hour=0)
+    end
+
+    if haskey(d, "Electrolyzer")
+        electrolyzer = MPCElectrolyzer(; dictkeys_tosymbols(d["Electrolyzer"])...)
+    else
+        electrolyzer = MPCElectrolyzer(; size_kw=0)
+    end
+
+    if haskey(d, "FuelCell")
+        fuel_cell = MPCFuelCell(; dictkeys_tosymbols(d["FuelCell"])...)
+    else
+        fuel_cell = MPCFuelCell(; size_kw=0)
+    end
+
+    if haskey(d, "HydrogenLoad")
+        hydrogen_load = MPCHydrogenLoad(; dictkeys_tosymbols(d["HydrogenLoad"])...)
+    else
+        hydrogen_load = MPCHydrogenLoad(; loads_kg = zeros(length(electric_load.loads_kw)))
+    end
+
+    if haskey(d, "Compressor")
+        compressor = MPCCompressor(; dictkeys_tosymbols(d["Compressor"])...)
+    else
+        compressor = MPCCompressor(; size_kw=0)
+    end
+
     # Placeholder/dummy cooling load set to zeros
```

Comment:
will these be actually implemented before merging or do we need to document and add errors/warnings?